src/test/operators/join_test_runner.cpp

```diff
@@ -0,0 +1,590 @@
+        join_test_configuration.nullable_left = true;
+        join_test_configuration.nullable_right = true;
+        join_test_configuration.predicate_condition = predicate_condition;
+
+        add_configuration_if_supported(join_test_configuration);
+      }
+    }
+
+    // The input tables are designed to have exclusive values. Test that these are handled correctly for different
+    // JoinModes by swapping the input tables.
+    // Additionally, go through all PredicateCondition to test especially the JoinSortMerge's different paths for these
+    for (const auto& predicate_condition : all_predicate_conditions) {
+      for (const auto& join_mode : all_join_modes) {
+        auto join_test_configuration = default_configuration;
+        join_test_configuration.join_mode = join_mode;
+        join_test_configuration.predicate_condition = predicate_condition;
+
+        join_test_configuration.swap_input_sides();
+
+        add_configuration_if_supported(join_test_configuration);
+      }
+    }
+
+    // Test all combinations of reference/data input tables. This tests mostly the composition of the output table
+    for (const auto table_size : {0u, all_table_sizes.front()}) {
+      for (const auto& left_input_table_type : all_input_table_types) {
+        for (const auto& right_input_table_type : all_input_table_types) {
+          auto join_test_configuration = default_configuration;
+          join_test_configuration.input_left.table_type = left_input_table_type;
+          join_test_configuration.input_left.table_size = table_size;
+          join_test_configuration.input_right.table_type = right_input_table_type;
+          join_test_configuration.input_right.table_size = table_size;
+
+          add_configuration_if_supported(join_test_configuration);
+        }
+      }
+    }
+
+    // Test multi-predicate jin support for all join modes. Swap the input tables to trigger cases especially in the
+    // Anti* modes where a secondary predicate evaluating to FALSE might "save" a tuple from being discarded
+    for (const auto& join_mode : all_join_modes) {
+      for (const auto& secondary_predicates : all_secondary_predicate_sets) {
+        for (const auto& predicate_condition : {PredicateCondition::Equals, PredicateCondition::NotEquals}) {
+          for (const auto swap_input_sides : {false, true}) {
+            auto join_test_configuration = default_configuration;
+            join_test_configuration.join_mode = join_mode;
+            join_test_configuration.secondary_predicates = secondary_predicates;
+            join_test_configuration.predicate_condition = predicate_condition;
+
+            if (swap_input_sides) {
+              join_test_configuration.swap_input_sides();
+            }
+
+            add_configuration_if_supported(join_test_configuration);
+          }
+        }
+      }
+    }
+
+    // Materialization phases might take advantage of certain encodings. JSM, e.g., has a special path for Dictionaries
```

Comment:
I think I've never seen the abbr. "JSM" before.

Reply:
finish with a dot.

Reply:
Strange, I use it all the time in my head. Changed to JoinSortMerge